eclair-core/src/main/scala/fr/acinq/eclair/transactions/Scripts.scala

```diff
@@ -297,4 +303,168 @@ object Scripts {
   def witnessHtlcWithRevocationSig(revocationSig: ByteVector64, revocationPubkey: PublicKey, htlcScript: ByteVector) =
     ScriptWitness(der(revocationSig) :: revocationPubkey.value :: htlcScript :: Nil)
 
+  /**
+   * Specific scripts for taproot channels
+   */
+  object Taproot {
+
+    import KotlinUtils._
+
+    implicit def scala2kmpscript(input: Seq[fr.acinq.bitcoin.scalacompat.ScriptElt]): java.util.List[fr.acinq.bitcoin.ScriptElt] = input.map(e => scala2kmp(e)).asJava
+
+    /**
+     * Sort and aggregate the public keys of a musig2 session
+     *
+     * @param pubkey1 public key
+     * @param pubkey2 public key
+     * @return the aggregated public key
+     * @see [[fr.acinq.bitcoin.scalacompat.Musig2.aggregateKeys()]]
+     */
+    def musig2Aggregate(pubkey1: PublicKey, pubkey2: PublicKey): XonlyPublicKey = Musig2.aggregateKeys(sort(Seq(pubkey1, pubkey2)))
+
+    /**
+     * "Nothing Up My Sleeve" point, for which there is no private key
+     */
+    val NUMS_POINT = PublicKey(ByteVector.fromValidHex("02dca094751109d0bd055d03565874e8276dd53e926b44e3bd1bb6bf4bc130a279"))
+
+    // miniscript: older(16)
+    val anchorScript: Seq[ScriptElt] = OP_16 :: OP_CHECKSEQUENCEVERIFY :: Nil
+
+    val anchorScriptTree = new ScriptTree.Leaf(anchorScript)
+
+    /**
+     * sends to a single revocation key
+     * miniscript: pk(revocation_key))
```

Comment:
this is the only significant change to the taproot channels proposal, where `toRevokeScript` is defined as:
```
<local_delayedpubkey> OP_DROP
<revocation_pubkey> OP_CHECKSIG
```
I don't see the point of using the `local_delayedpubkey` here but maybe I've missed something...

Reply:
> ...: if we want to directly use the revocation key, then we should put it in the `internal_key` and use the key path spend for this instead of using a script, ...

I thought about this but it makes sense when the `key path` spend is the most common case, here it's the opposite. But it's worth mentioning.

I still don't see the point of having the  `local_delayedpubkey` revealed during a revoke spent (if the spender is able to monitor the blockchain and compute the correct revocation key they should be able to also recompute the correct local  delayed key) but I agree that it's better to revert to the original script for now.

Reply:
I reverted to the original `to-revoke script` in https://github.com/ACINQ/eclair/pull/3016/commits/971ee5a3c6e78e5aaa9ac4db67c503269a29c17f. 

I've also modified the unit test to highlight the problem here: anchor outputs can be spend without a signature after the 16-blocks CSV delay BUT spenders need to know the local/remote payment public key.

Reply:
Yes that's why I added how to spend the anchor outputs in the unit tests. I thought I had found a way to make them spendable by anyone after 16 confirmations,  with just the information available in the published commit tx, but I was wrong :(